source/routing/raptor.cpp

```diff
@@ -62,6 +62,10 @@ bool RAPTOR::apply_vj_extension(const Visitor& v, const bool global_pruning,
     bool add_vj = false;
     bool result = false;
     while(vj) {
+        if (! vj->validity_pattern->check(DateTimeUtils::date(workingDt))) {
```

Comment:
It should be tested with is_valid_day 6 lines behind. And you don't check adatped if needed.